Wikipedia/Code/Article as a Living Document/ArticleAsLivingDocHorizontallyScrollingCell.swift

```diff
@@ -0,0 +1,87 @@
+
+import UIKit
+
+protocol ArticleAsLivingDocHorizontallyScrollingCellDelegate: class {
+    func tappedLink(_ url: URL, cell: ArticleAsLivingDocHorizontallyScrollingCell?, sourceView: UIView, sourceRect: CGRect?)
+}
+
+class ArticleAsLivingDocHorizontallyScrollingCell: CollectionViewCell {
+    let descriptionTextView = UITextView()
+    var theme: Theme?
+    
+    weak var delegate: ArticleAsLivingDocHorizontallyScrollingCellDelegate?
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    override func reset() {
+        super.reset()
+        descriptionTextView.attributedText = nil
+    }
+    
+    override func sizeThatFits(_ size: CGSize, apply: Bool) -> CGSize {
+        fatalError("Must override sizeThatFits in subclass")
+    }
+    
+    func configure(change: ArticleAsLivingDocViewModel.Event.Large.ChangeDetail, theme: Theme, delegate: ArticleAsLivingDocHorizontallyScrollingCellDelegate) {
+        
+        setupDescription(for: change)
+        updateFonts(with: traitCollection)
+        
+        backgroundView?.layer.cornerRadius = 3
+        backgroundView?.layer.masksToBounds = true
+        selectedBackgroundView?.layer.cornerRadius = 3
+        selectedBackgroundView?.layer.masksToBounds = true
+        layer.shadowOffset = CGSize(width: 0, height: 2)
+        layer.shadowOpacity = 0.3
+        layer.shadowRadius = 3
+        
+        apply(theme: theme)
+        self.delegate = delegate
+    }
+    
+    func setupDescription(for change: ArticleAsLivingDocViewModel.Event.Large.ChangeDetail) {
+        
+        let description: NSAttributedString
+        switch change {
+        case .snippet(let snippet):
+            description = snippet.displayText
```

Comment:
Not a bug, but could be compacted if this and two lines below are `descriptionTextView.attributedText = reference.description`

Reply:
Updated